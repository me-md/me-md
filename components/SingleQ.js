import React, { useState, Fragment } from 'react';
import { Dimensions, StyleSheet, View } from 'react-native';
import { Body, Button, Card, Text } from 'native-base';
import { AntDesign } from '@expo/vector-icons';

const height = Dimensions.get('window').height;

export default function SingleQ({ question, answerQuestion }) {
  const [checkWho, setCheckWho] = useState([]);

  const handleSubmit = () => {
    answerQuestion([
      {
        id: question.question.items[0].id,
        choice_id: checkWho
      }
    ]);
    setCheckWho([]);
  };

  const choices = question.question.items[0].choices.map((choice, index) => {
    switch (choice.label) {
      case 'Yes':
        return (
          <Fragment key={index}>
            <Text>Yes</Text>
            <AntDesign
              key={index}
              name={checkWho === choice.id ? 'checkcircle' : 'checkcircleo'}
              id={choice.id}
              style={styles.add}
              size={32}
              color='green'
              onPress={() => {
                setCheckWho(choice.id);
              }}
            />
          </Fragment>
        );
      case 'No':
        return (
          <Fragment key={index}>
            <Text>No</Text>
            <AntDesign
              key={index}
              name={checkWho === choice.id ? 'closecircle' : 'closecircleo'}
              id={choice.id}
              style={styles.add}
              size={32}
              color='red'
              onPress={() => {
                setCheckWho(choice.id);
              }}
            />
          </Fragment>
        );
      case `Don't know`:
        return (
          <Fragment key={index}>
            <Text>Unsure</Text>
            <AntDesign
              key={index}
              name={
                checkWho === choice.id ? 'questioncircle' : 'questioncircleo'
              }
              id={choice.id}
              style={styles.add}
              size={32}
              color='black'
              onPress={() => {
                setCheckWho(choice.id);
              }}
            />
          </Fragment>
        );
      default:
        <></>;
    }
  });

  return (
    <Card id={question.question.items[0].id} style={styles.questionCard}>
      <Body style={styles.cardBody}>
        <Text style={styles.questionText}>{question.question.text}</Text>
        <View style={styles.checkboxes}>{choices}</View>
        {checkWho.length ? (
          <Button style={styles.button} block onPress={() => handleSubmit()}>
            <Text style={styles.buttonText}>SUBMIT</Text>
          </Button>
        ) : (
          <Button style={styles.button} disabled block>
            <Text style={styles.buttonText}>SUBMIT</Text>
          </Button>
        )}
      </Body>
    </Card>
  );
}

const styles = StyleSheet.create({
  questionCard: {
    backgroundColor: '#f7f7f7',
    marginBottom: height * 0.01,
    marginTop: height * 0.01,
    shadowColor: 'black',
    shadowOffset: { width: 5, height: 5 },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    width: '80%'
  },
  cardBody: {
    padding: height * 0.02,
    width: '100%'
  },
  questionText: {
    alignSelf: 'flex-start',
    flex: 0.5,
    fontSize: 16,
    marginBottom: height * 0.02,
    marginTop: height * 0.01,
    padding: 0
  },
  checkboxes: {
    alignItems: 'center',
    flexDirection: 'row',
    fontSize: 12,
    justifyContent: 'space-around',
    marginBottom: height * 0.02,
    padding: 0,
    width: '90%'
  },
  button: {
    shadowColor: 'black',
    shadowOffset: { width: 5, height: 5 },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
  },
  buttonText: {
    fontWeight: 'bold'
  }
});
