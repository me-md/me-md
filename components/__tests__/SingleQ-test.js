import React from 'react';
import renderer from 'react-test-renderer';

import SingleQ from '../SingleQ';

it(`should render correctly`, () => {
  const mockAnswerQuestion = jest.fn();
  const mockQuestion = {
    question: {
      text: 'Where\'s Rachel?!?',
      items: [
        {
          id: 0,
          choices: [
            {
              label: 'Yes',
              id: 'present'
            },
            {
              label: 'No',
              id: 'absent'
            },
            {
              label: 'unknown',
              id: 'Don\'t Know'
            }
          ]
        },

      ]
    }
  }

  const tree = renderer.create(<SingleQ question={mockQuestion} answerQuestion={mockAnswerQuestion} />).toJSON();

  expect(tree).toMatchSnapshot();
});