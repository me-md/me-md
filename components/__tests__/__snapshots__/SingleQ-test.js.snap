// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  id={0}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "backgroundColor": "#f7f7f7",
        "marginBottom": 13.34,
        "marginTop": 13.34,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
        "width": "80%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        },
        Object {
          "padding": 26.68,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "alignSelf": "flex-start",
            "flex": 0.5,
            "fontSize": 16,
            "marginBottom": 26.68,
            "marginTop": 13.34,
            "padding": 0,
          },
        ]
      }
      uppercase={false}
    >
      Where's Rachel?!?
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "fontSize": 12,
          "justifyContent": "space-around",
          "marginBottom": 26.68,
          "padding": 0,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Yes
      </Text>
      <Text />
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        No
      </Text>
      <Text />
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#b5b5b5",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 57,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 93.38000000000001,
          "justifyContent": "space-around",
          "marginTop": 66.7,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4.65,
          "width": 487.5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "fontSize": 33.35,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
      >
        Continue
      </Text>
      <Text />
    </View>
  </View>
</View>
`;
